lib/ansiblelint/rules/MissingFilePermissionsRule.py

```diff
@@ -19,57 +19,77 @@
 # THE SOFTWARE.
 from ansiblelint.rules import AnsibleLintRule
 
+# Despite documentation mentioning 'preserve' only these modules support it:
+_modules_with_preserve = (
+    'copy',
+    'template',
+)
+
 
 class MissingFilePermissionsRule(AnsibleLintRule):
     id = "208"
-    shortdesc = 'File permissions not mentioned'
+    shortdesc = 'File permissions unset or incorrect'
     description = (
-        "Missing mode parameter can cause unexpected file permissions based "
-        "on version of Ansible being used. Be explicit, or if you still "
-        "want the default behavior you can use ``mode: preserve`` to avoid "
-        "hitting this rule. See "
-        "https://github.com/ansible/ansible/issues/71200"
+        "Missing or unsupported mode parameter can cause unexpected file "
+        f"permissions based "
+        f"on version of Ansible being used. Be explicit, like ``mode: 0644`` to "
+        f"avoid hitting this rule. Special ``preserve`` value is accepted "
+        f"only by {', '.join(_modules_with_preserve)} modules. "
+        f"See https://github.com/ansible/ansible/issues/71200"
     )
     severity = 'VERY_HIGH'
     tags = ['unpredictability', 'experimental']
     version_added = 'v4.3.0'
 
     _modules = (
-        'assemble',
         'archive',
-        'copy',
+        'assemble',
+        'copy',  # supports preserve
         'file',
-        'template',
+        'replace',  # implicit preserve behavior but mode: preserve is invalid
+        'template',  # supports preserve
         'unarchive',
     )
 
-    _modules_with_create = (
-        'blockinfile',
-        'ini_file',
-        'lineinfile'
-    )
+    _modules_with_create = {
+        'blockinfile': False,
+        'ini_file': True,
+        'lineinfile': False,
+        'htpasswd': True,
```

Comment:
ok htpasswd apprently supports preserve from 2.6: https://docs.ansible.com/ansible/latest/modules/htpasswd_module.html#parameter-mode

don't you keep these lists in alphabetical order ?

Reply:
The documentation of Ansible 2.9 is incorrect. The `htpasswd` module does not support `preserve`. 
I have verified this myself and created PRs for fixing the docs. You can read more on this problem here: https://github.com/ansible/ansible/issues/56839